src/Stan/Config.hs

```diff
@@ -207,10 +219,111 @@ applyChecks paths = foldl' useCheck (mkDefaultChecks paths)
     applyForScope f mScope hm = case mScope of
         Nothing -> f <$> hm  -- no scope = apply for everything
         Just cScope -> case cScope of
-            CheckScopeFile path -> HashMap.alter (fmap f) path hm
+            CheckScopeFile path -> HashMap.adjust f path hm
             CheckScopeDirectory dir -> HashMap.mapWithKey
                 (\path -> if isInDir dir path then f else id)
                 hm
 
     isInDir :: FilePath -> FilePath -> Bool
     isInDir dir path = dir `isPrefixOf` path
+
+{- $applyConfig
+
+The 'applyConfig' function transforms the list of rules defined in the
+'Config' (either via TOML or CLI) to get the list of 'Inspection's for
+each module.
+
+By default, @stan@ runs all 'Inspection's for all modules in the
+Haskell project and outputs all 'Observation's it found. Using
+'Config', you can adjust the default setting using your preferences.
+
+=== Algorithm
+
+The algorithm for figuring out the resulting set of 'Inspection's per
+module applies each 'Check' one-by-one in order of their appearance.
+
+When introducing a new 'Check' in the config, you must always specify
+the 'CheckType' (either 'Include' or 'Ignore'). 'CheckFilter' and
+'CheckScope' can be omitted. If they are not written explicitly, then
+the 'Check' applies to all entries. Specifically:
+
+* If 'CheckFilter' is not specified, 'Check' applies to all
+  inspections for the given 'CheckScope' (either 'Ignore' or 'Include'
+  all 'Inspection's)
+* If 'CheckScope' is not specified, 'Check' applies given
+  'CheckFilter' to all files
+
+As a result of this approach, when both 'CheckFilter' and 'CheckScope'
+are not specified explicitly, either all 'Inspection's are 'Ignore'd,
+or all inspections are 'Include'd back.
+
+The algorithm doesn't remove any files or inspections from the
+consideration completely. So, for example, if you ignore all
+inspections in a specific file, new inspections can be added for this
+file later by the follow up rules. If you want to ignore some files or
+directories completely, put 'Check's for them at the end of your
+configuration.
```

Comment:
What if could introduce another type of check for that? I see how it could be useful to completely remove the file from consideration (e.g. generated files)

Reply:
@vrom911 As discussed, issue #158 was created to address this problem.